### PR TITLE
Refactor activity log sources/views/indexes

### PR DESCRIPTION
**Pull request by umanwizard**
This PR does the following:

* Gets rid of `mz_activity_log` and `mz_activity_log_redacted`. These are basically pure footguns that nobody should be using. Instead, folks should use `mz_recent_activity_log` and `mz_recent_activity_log_redacted`.
* Moves the SQL / redacted SQL text of queries out to a source called `mz_sql_text`. This allows identical SQL to be deduplicated.
* Removes the SQL text and redacted SQL text from `mz_prepared_statement_history`.
* Incorporates the session history into `mz_recent_activity_log` and `mz_recent_activity_log_redacted`.
* Instead of indexing `mz_recent_activity_log`, index its two inputs: the SQL text and everything but the SQL text. These are in the (undocumented) views `mz_recent_activity_log_thinned` and `mz_recent_sql_text`. That way, the amount of data we have to index is much smaller (due to deduplication in `mz_sql_text`.

After this rolls out, before it goes to prod, we need to do the following:

* Update the console to change its queries to not join in `mz_session_history`, which is no longer necessary.
* For users who were using the old `mz_activity_log` or `mz_activity_log_redacted`, have them switch to the newer `recent` versions if possible
* For users who were using the underlying `mz_prepared_statement_history` source, make sure they know they now need to also join against `mz_sql_text` (on `sql_hash`) to get the full SQL text.

### CODE REVIEW

**Review by mjibson — APPROVED** [review]

File: src/catalog/src/builtin.rs [hunk 1]
```diff
@@ -2547,15 +2547,46 @@ pub static MZ_PREPARED_STATEMENT_HISTORY: Lazy<BuiltinSource> = Lazy::new(|| Bui
...
+    sql: "SELECT DISTINCT sql_hash, sql, redacted_sql FROM mz_internal.mz_sql_text WHERE prepared_day + INTERVAL '4 days' >= mz_now()",
+    access: vec![MONITOR_SELECT],
+}
+});
+
+pub static MZ_RECENT_SQL_TEXT_REDACTED: Lazy<BuiltinView> = Lazy::new(|| BuiltinView {
+    name: "mz_recent_sql_text_redacted",
+    schema: MZ_INTERNAL_SCHEMA,
+    column_defs: None,
+    sql: "SELECT sql, redacted_sql FROM mz_internal.mz_recent_sql_text",
```
[the remark this answers is not in the document]
Comment (umanwizard) [reply src/catalog/src/builtin.rs:2581]: woah, good catch. Would have been bad to land that!

File: src/adapter/Cargo.toml [hunk 2]
```diff
@@ -31,6 +31,7 @@ launchdarkly-server-sdk = { version = "1.0.0", default_features = false, feature
   "hypertls",
 ] }
 maplit = "1.0.2"
+md-5 = "0.10.5"
```
[the remark this answers is not in the document]
Comment (umanwizard) [reply src/adapter/Cargo.toml:34]: My idea was to start out with md5 and then if we decide we can handle the extra bytes of memory, upgrade to sha256, which will cost us an extra 16 bytes for every element in `mz_sql_text` and `mz_prepared_statement_history`. I think md5 is fine though for now, as I find it hard to imagine a scenario where preventing collisions here is security-critical. I'd be willing to cede on that if someone feels strongly.

File: src/catalog/src/builtin.rs [hunk 3]
```diff
@@ -2547,15 +2547,46 @@ pub static MZ_PREPARED_STATEMENT_HISTORY: Lazy<BuiltinSource> = Lazy::new(|| Bui
...
+    sql: "SELECT sql_hash, redacted_sql FROM mz_internal.mz_sql_text",
+    access: vec![MONITOR_SELECT, MONITOR_REDACTED_SELECT, SUPPORT_SELECT],
+});
+
+pub static MZ_RECENT_SQL_TEXT: Lazy<BuiltinView> = Lazy::new(|| {
+    BuiltinView {
+    name: "mz_recent_sql_text",
+    schema: MZ_INTERNAL_SCHEMA,
+    column_defs: None,
+    sql: "SELECT DISTINCT sql_hash, sql, redacted_sql FROM mz_internal.mz_sql_text WHERE prepared_day + INTERVAL '4 days' >= mz_now()",
```
[the remark this answers is not in the document]
Comment (umanwizard) [reply src/catalog/src/builtin.rs:2572]: Yes, I'll add a comment.

File: src/catalog/src/builtin.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjibson) [reply src/catalog/src/builtin.rs:2581]: Can you add a test that would have caught this?

[the remark this answers is not in the document]
Comment (umanwizard) [reply src/catalog/src/builtin.rs]: Done

**Pull request merged**